sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs

```diff
@@ -251,7 +251,7 @@ private List<NonInteractiveAzureServiceTokenProviderBase> GetTokenProviders()
 
         public async Task<string> GetAccessTokenAsync(string resource, string tenantId)
         {
-            return await GetAccessTokenAsync(resource, tenantId, default(CancellationToken));
+            return await GetAccessTokenAsync(resource, tenantId, default(CancellationToken)).ConfigureAwait(false);
```

Comment:
Remove the async await. There is no reason for it here.

Reply:
Generally you are correct that this would be possible and would remove the necessity to have the context capture configuration. Usually these kinds of optimizations though only make sense in high throughput code. The new best practices are to write the await statements, see

https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md

Plus it's aligned with how the code was before